### PR TITLE
feat(webapp):  set different icon and text on join leave button, when member is pending

### LINKED ISSUE #6014: 🚀 [Feature] Webapp – Join / Leave Button on Group Page is Misleading
## 🚀 Feature

Webapp – Join / Leave Button on Group Page is Misleading.

### Problem

If I join a closed group and become a pending member the button states, that I am a meber already.
Only underneath in my role I can see that I am pending. That's not clear enough.

### Todo

- if I'm a pending member change the join/leave button title to `membership penting`:
  - EN: `I'm a member` → `Pending member` (or something a bit shorter)
  - DE `Bin Mitglied` → `Ausstehendes Mitglied` (or something a bit shorter)
- additional change the role tranlations of the role `usual` to:
  - EN: `Simple Member` → `Member`
  - DE `Einfaches Mitglied` → `Mitglied`

### PR DESCRIPTION
**Pull request by mahula**
## 🍰 Pullrequest
Make the `JoinLeaveButton` on a group page unambiguous for pending members

### Issues
- fixes #6014 

### Todo
- [x] for a pending member add new key to locales and set it as button text
- [x] change role tranlations of the role `usual`
  - EN: `Simple Member` → `Member`
  - DE `Einfaches Mitglied` → `Mitglied`
- [x] Specify the final icon to use @Tirokk 
  - [x] `(?)` is cool
- [x] change `√ I'm a member` on hover to `X Leave` instead of `X I'm a member`
- [x] on `(?) Pending member` have a tooltip:
  - [x] EN: `The owner has yet to confirm you.`
  - [x] DE: `Der Inhaber muss dich noch bestätigen.`

### CODE REVIEW

File: webapp/components/Button/JoinLeaveButton.vue [hunk 1]
```diff
@@ -35,15 +36,24 @@ export default {
...
-      if (this.isMember && this.hovered) {
-        return 'close'
+      if (this.isMember) {
+        if (this.hovered) {
+          return 'close'
+        } else if (this.isNonePendingMember) {
+          return 'check'
+        } else {
+          return 'question-circle'
+        }
```
Comment (Elweyn) [inline webapp/components/Button/JoinLeaveButton.vue:46]: I don't like if their is so much depth of if else..
Maybe change the multiple return with a variable that is returned at the end.
```suggestion
      if (this.isMember && this.hovered) {
        return 'close'
      } else if (this.isMember && this.isNonePendingMember) {
        return 'check'
      } else if (this.isMember) {
        return 'question-circle'
```

File: webapp/components/Button/JoinLeaveButton.vue [hunk 2]
```diff
@@ -35,15 +36,24 @@ export default {
...
+        return 'plus'
       }
     },
     label() {
       if (this.isMember) {
-        return this.$t('group.joinLeaveButton.iAmMember')
+        if (this.isNonePendingMember) {
+          return this.$t('group.joinLeaveButton.iAmMember')
+        }
+        return this.$t('group.joinLeaveButton.pendingMember')
```
Comment (Elweyn) [inline webapp/components/Button/JoinLeaveButton.vue:56]: Could do something similar with this.

**Review by Elweyn — COMMENTED** [review]
I dont like it if their is to much if else blocks depth. 
But I am ready to debate if it is necessary.

File: webapp/components/Button/JoinLeaveButton.vue (hunk 1 above)
Comment (mahula) [reply webapp/components/Button/JoinLeaveButton.vue:46]: I know, on the first glimpse,it looks a bit much.
For me that clearly depicted the logic - which gets you quicker to a decision bewteen `isMember` being `true` or `false`.
The second if/else level is taking care of the three cases, if `isMember` is `true`.
Amnot sure, how much a variable would help the logic.

File: webapp/locales/de.json [hunk 3]
```diff
@@ -447,7 +447,10 @@
     "in": "in",
     "joinLeaveButton": {
       "iAmMember": "Bin Mitglied",
-      "join": "Beitreten"
+      "join": "Beitreten",
+      "leave": "Verlassen",
+      "pendingMember": "Ausstehendes Mitglied",
+      "tooltip": "Der Inhaber muss dich noch bestätigen."
```
Comment (Tirokk) [inline webapp/locales/de.json:453]:
```suggestion
      "tooltip": "Der Gruppeninhaber muss dich noch bestätigen."
```

**Review by Tirokk — CHANGES_REQUESTED** [review]
I showed @LarsE11 your code and our ideas and he really likes it.

Just one little suggestion …

**Review by Tirokk — APPROVED** [review]
YEAH ! ❣️

I approve 🚀💫

**Pull request merged**